integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/classMediator/PropertyPersistenceDeletingAndAddingTstCase.java

```diff
@@ -83,29 +89,28 @@ public void testMediationPersistenceDeletingAndAdding() throws Exception {
         Deleting User and email   refer: https://wso2.org/jira/browse/TA-532           param 5
          */
 
+            serverConfigurationManager.removeFromComponentLib(CLASS_JAR_FIVE_PROPERTIES);
+            serverConfigurationManager.copyToComponentLib
+                    (new File(getClass().getResource(JAR_LOCATION + File.separator + CLASS_JAR_FOUR_PROPERTIES).toURI()));
+            loadSampleESBConfiguration(0);
+            /* waiting for the new config file to be written to the disk */
+            Thread.sleep(10000);
```

Comment:
Shall we look into a way to remove this thread sleep? Maybe using something like [Awaitility](https://github.com/awaitility/awaitility)?

Reply:
Changed. Please refer latest commit